src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs

```diff
@@ -56,16 +54,21 @@ public bool TryRegister(SafeSocketHandle socket, out Interop.Error error)
 
         private static readonly object s_lock = new object();
 
-        // In debug builds, force there to be 2 engines. In release builds, use half the number of processors when
-        // there are at least 6. The lower bound is to avoid using multiple engines on systems which aren't servers.
-#pragma warning disable CA1802 // const works for debug, but needs to be static readonly for release
-        private static readonly int s_engineCount =
+        private static readonly int s_engineCount = GetEnginesCount();
+
+        private static int GetEnginesCount()
+        {
+            if (uint.TryParse(Environment.GetEnvironmentVariable("SYSTEM_NET_SOCKETS_ENGINE_COUNT"), out uint count))
```

Comment:
Can we rename `SYSTEM_NET_SOCKETS_ENGINE_COUNT` to `DOTNET_SYSTEM_NET_SOCKETS_THREAD_COUNT`?  We've been using the DOTNET_ prefix in general, and "thread" is more meaningful than "engine".